tests/cypress/page-object/pageComposer.ts

```diff
@@ -146,11 +151,39 @@ export class PageComposer extends BasePage {
 
     navigateToPage(name: string): PageComposer {
         cy.iframe('#page-composer-frame', this.iFrameOptions).within(() => {
-            cy.get('#JahiaGxtPagesTab').contains(name).click({force: true});
+            cy.get('#JahiaGxtPagesTab').find('span').contains(name).click({force: true});
         });
 
         return new PageComposer();
     }
+
+    switchLanguage(language: string) {
+        cy.iframe('#page-composer-frame', this.iFrameOptions).within(() => {
+            cy.get('.toolbar-itemsgroup-languageswitcher').click();
+            cy.get('.x-combo-list-item').contains(language).click();
+        });
+    }
+
+    doInsideInnerFrame(fcn: () => void) {
+        cy.iframe('#page-composer-frame', this.iFrameOptions).within(() => {
+            cy.iframe('.gwt-Frame', this.iFrameOptions).within(() => {
+                fcn();
+            });
+        });
+    }
+
+    publish(menuEntry: string, selectorText: string) {
+        cy.iframe('#page-composer-frame', this.iFrameOptions).within(() => {
+            cy.get('.edit-menu-publication').click();
+            cy.get('.menu-edit-menu-publication').find('span').contains(menuEntry).click();
+            cy.get('button').contains(selectorText).click();
+        });
+    }
+
+    checkIfPublished() {
+        /* eslint-disable cypress/no-unnecessary-waiting */
+        cy.wait(3000);
```

Comment:
any way to actually check if it's published? maybe use wait-until and query the live workspace until node appears? if not, other check I do is do a retry on the live site to wait for component to be published.